checker-qual/src/main/java/org/checkerframework/dataflow/qual/SideEffectsOnly.java

```diff
@@ -0,0 +1,28 @@
+package org.checkerframework.dataflow.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.checkerframework.framework.qual.JavaExpression;
+
+/**
+ * This method declaration annotation can be used to specify the expressions that a method side
+ * effects. In other words, the method only side effects those expressions that are supplied as
+ * annotation values to {@code @SideEffectsOnly}.
```

Comment:
I think `@SideEffectsOnly({})` is equivalent to `@SideEffectFree`.  That might be worth stating here, or it might be more appropriate in the manual which already remarks on the relationship between the two annotations.